### DISCUSSION TITLE
How to know which version is/isn't stable?

### DISCUSSION THREAD

**Discussion by chrisweis** [Snowflake-Labs/terraform-provider-snowflake#2407] [Q&A]
Hi,

I've been using this Snowflake Terraform Provider for a while now, but I'm struggling to know which versions are stable or unstable as I aim to upgrade.  I want to avoid upgrading to a version that's buggy, and instead hop periodically to each of the more stable versions.  Is there a way to determine that without rummaging through each open Github Issue, of which there seem to be many?

Thanks,
Chris

**Comment by sfc-gh-asawicki** [comment]
Hey @chrisweis. Thanks for the question.

As we are still in the 0.x.x version, there is no stability guarantee in any of the existing versions. As stated in the [disclaimer in our README](https://github.com/Snowflake-Labs/terraform-provider-snowflake?tab=readme-ov-file#snowflake-terraform-provider), we encourage you always to use the newest version. We periodically improve the provider by moving away from the old buggy implementation and resolving existing issues.

This will change when we reach V1, our next bigger milestone.

As you pointed out there are many open issues. We have already categorized them and we plan to focus on them shortly. You can read more about the issues in this [ROADMAP overview](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/ROADMAP.md#19012024-roadmap-overview) (specifically [here](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/ROADMAP.md#resolving-existing-issues)).

**Comment by chrisweis** [comment]
Hi @sfc-gh-asawicki,

Thanks for the quick reply!  I’d seen that note about the stability already, and probably didn’t phrase my question clearly.  I appreciate the challenge of getting to v1 and I’m more asking how we can know which versions are the **most stable** historically.  I probably won’t frequently update us to the latest version and instead crave predictability.

For example, might we be able to query the GitHub Issues by version number as a heuristic to roughly approximate stability?  I tend to lag behind in upgrading to allow for choosing the better versions, but don’t know which of those versions are worse or better.

Even potentially a markdown page/table generated via ChatGPT with “Stable” or “Unstable” for each version, based upon a dump of all issues and comments would be immensely helpful.  I’d be willing to contribute some API money to fund that.

Thanks for the discussion,
Chris

**Comment by sfc-gh-asawicki** [comment]
Hey @chrisweis. This is an interesting idea; we will give it a thought!

As for now, you can consider the currently existing versions after v0.76.0 to be the stable ones (there were some problems between v0.73.0-v0.76.0). However, again, they are not free of problems, so it also depends on the resources you are using.
